haystack/schema.py

```diff
@@ -466,6 +466,7 @@ def __eq__(self, other):
         return (
             isinstance(other, self.__class__)
             and is_content_equal
+            and getattr(other, "answer", None) == self.answer
```

Comment:
@ZanSara @silvanocerza This is the only change to the main code I made. I believe it makes sense that for checking equality between two `Answer` objects we should be checking the answer text, but for some reason, it was not there before. This is a breaking change since it will change the behavior of how answer objects are compared.

Any ideas why we should not have add this?

Reply:
I would consider this a bug fix. 😅

Reply:
If there are no existing tests breaking due to this change I'd also consider it a bugfix instead of a breaking change

Reply:
I thought it would be a breaking change because it could cause downstream codes to break since it changes the default behavior `Answer`.

Reply:
And to be fair, if they're untested they deserve to break :sweat_smile: Tests are written exactly to remove this kind of guesswork.

Reply:
Sorry, I meant other code bases that rely on Haystack.

Reply:
Alright we can keep the label then 👍